lib/smart_answer/calculators/state_pension_age_calculator.rb

```diff
@@ -16,6 +16,19 @@ def state_pension_date
       StatePensionDateQuery.state_pension_date(dob, gender)
     end
 
+    def approaching_pension_age?
+      difference = (state_pension_date - Date.today).to_i
+      if difference >= 0 && difference <= claim_can_start_within
+        true
+      else
+        false
+      end
+    end
+
+    def claim_can_start_within
+      (4.month.since.to_date - Date.today).to_i + 4
+    end
```

Comment:
Yeah but the intention I had was to 4months and 4 days from the current day that the user is actually using the smart answer. `claim_can_start_within ` returns the number of days from today. 

For months like February that are not up to 31 days I can easily return the right number of days + 4 from the day the user uses the state pension age smart answer.  

with what you have here.. `state_pension_date - 4.months - 4.days` it is basd on the proposed date when they reach state pension age,

The purpose for this is to encourage users who will be reaching state pension ages within 4 months and 4 days time to apply